File: common/src/main/java/com/resourcify/common/model/NotificationMessage.java
package com.resourcify.common.model;

import com.resourcify.common.model.enums.StatusEnum;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class NotificationMessage {

  @NotNull
  @NotEmpty
  private String forUsername;
  @NotNull
  @NotEmpty
  private String message;

  private StatusEnum status;
}

File: frontend-angular/src/app/model/NotificationMessage.ts
export type NotificationMessage = {
  forUsername: string;
  message: string;
  status: Status;
}

export enum StatusEnum {
  APPROVED = "APPROVED",
  DECLINED = "DECLINED",
  PENDING = "PENDING",
}

export type Status = StatusEnum.APPROVED | StatusEnum.DECLINED | StatusEnum.PENDING;
